clis/teliod/src/main.rs

```diff
@@ -115,17 +152,188 @@ impl CommandListener {
     }
 }
 
+async fn init_api(
+    config: &mut TeliodDaemonConfig,
+    config_path: String,
+) -> Result<ClientConfig, TeliodError> {
+    let mut client_config = ClientConfig::new(config).await?;
+
+    if client_config.machine_identifier.eq("") {
+        client_config.machine_identifier =
+            match load_identifier_from_api(&config.authentication_token, client_config.public_key)
+                .await
+            {
+                Ok(id) => id,
+                Err(e) => {
+                    debug!("Unable to load identifier due to {e}. Registering ...");
+                    register_machine(
+                        &client_config.hw_identifier,
+                        client_config.public_key,
+                        &client_config.auth_token,
+                    )
+                    .await?
```

Comment:
I'm not sure if it makes sense to try to register after any kind of failure - when it is `FetchingIdentifierError` then sure, but if you encountered some connectivity issue I guess it would be better to make some retries, as you do with update? And tbh to register the same applies

Reply:
Applied exp backoff to register and fetching identifier